VeganVirus/VeganVirus/ExcelAction.cpp

```diff
@@ -0,0 +1,128 @@
+#include "ExcelAction.h"
+
+
+
+BOOL CALLBACK windowCountCallback(
+    HWND hwnd,
+    LPARAM lParam
+)
+{
+    WindowCountCallbackParams& params = *(WindowCountCallbackParams*)lParam;
+
+    char title[100] = { 0 };
+    GetWindowTextA(hwnd, title, 100);
+    char* ret;
+
+    // check if title contains params.searchTerm
+    ret = strstr(title, params.searchTerm);
+    if (ret != NULL) {
+        params.count++;
+    }
+    return TRUE;
+}
+ExcelAction::ExcelAction(double req) : Action(req, EXCEL_ICON_PATH)
+{
+}
+void ExcelAction::start()
+{
+    _preProcessCount = excelWindowCount();
+    this->activeFlag = true;
+}
+void ExcelAction::update(double dt)
+{
+    if (!this->activeFlag)
+    {
+        return;
+    }
+    int currPocessCount = excelWindowCount();;
+    if (currPocessCount > _preProcessCount)
+    {
+        createFile(num_file);
+        openFile(num_file);
+        _vectorExcelFiles.insert(num_file);
+        num_file += 1;
+        currPocessCount++;
+        Sleep(1000);
+    }
+    else if (currPocessCount < _preProcessCount)
+    {
+        deleteAllExcelFile(_vectorExcelFiles);
+    }
+    _preProcessCount = currPocessCount;
+}
+/*!
+\brief Check if a process is running
+\param [in] processName Name of process to check if is running
+\returns \c True if the process is running, or \c False if the process is not running
+*/
+int ExcelAction::processCount(const char* searchTerm)
+{
+    WindowCountCallbackParams params = { 0, searchTerm };
+    EnumWindows(windowCountCallback, (LPARAM)&params);
+    return params.count;
+
+}
+/*!
+    opens file
+    param int num of the file
+    return None
+*/
+void ExcelAction::openFile(int num)
+{
+    std::string fileToOpen = std::to_string(num) + EXCEL_FILE_EXTENSION;
+    system(fileToOpen.c_str());
+
+}
+/*!
+    create file
+    param int num of the file
+    return None
+*/
+void ExcelAction::createFile(int num)
+{
+    std::string fileToCreat = std::to_string(num) + EXCEL_FILE_EXTENSION;
+    bool ret = CopyFileA("liron.xlsx", fileToCreat.c_str(), TRUE);
```

Comment:
Should add a define for the excel document to open. Maybe `#define EXCEL_DOCUMENT_PATH`